library/core/src/intrinsics/mod.rs

```diff
@@ -3314,3 +3314,27 @@ pub unsafe fn va_arg<T: VaArgSafe>(ap: &mut VaListImpl<'_>) -> T;
 #[rustc_intrinsic]
 #[rustc_nounwind]
 pub unsafe fn va_end(ap: &mut VaListImpl<'_>);
+
+/// Compiler-generated move operation - never actually called.
+/// Used solely for profiling and debugging visibility.
+///
+/// This function serves as a symbolic marker that appears in stack traces
+/// when rustc generates move operations, making them visible in profilers.
+/// The SIZE parameter encodes the size of the type being moved in the function name.
+#[rustc_force_inline]
+#[rustc_diagnostic_item = "compiler_move"]
+pub fn compiler_move<T, const SIZE: usize>(_src: *const T, _dst: *mut T) {
+    unreachable!("compiler_move should never be called - it's only for debug info")
+}
+
+/// Compiler-generated copy operation - never actually called.
+/// Used solely for profiling and debugging visibility.
+///
+/// This function serves as a symbolic marker that appears in stack traces
+/// when rustc generates copy operations, making them visible in profilers.
+/// The SIZE parameter encodes the size of the type being copied in the function name.
+#[rustc_force_inline]
+#[rustc_diagnostic_item = "compiler_copy"]
+pub fn compiler_copy<T, const SIZE: usize>(_src: *const T, _dst: *mut T) {
+    unreachable!("compiler_copy should never be called - it's only for debug info")
+}
```

Comment:
These aren't intrinsics so I don't think this is the best place for them. The file is already too big anyway.^^

Reply:
Yeah I wasn't sure exactly where to put them. Originally I had the idea of actually making them real functions implementing copy & move in terms of calls to them, but that seemed more fiddly than it's worth.